File: python/src/adapter/driven/sms_gateway_adapter.py
import logging

from src.domain.sms_gateway import SmsGateway

logger = logging.getLogger(__name__)


class SmsGatewayAdapter(SmsGateway):
    """短信接口Adapter"""

    async def send_code(self, mobile: str, code: str) -> None:
        logger.info(f"发送手机验证码到 {mobile}, code:  {code}")

File: dotnet/src/adapter/driven/SmsGatewayAdapter.cs
using BackendExample.Domain;
using log4net;

namespace BackendExample.Adapter.Driven
{
    /// <summary>
    /// 短信 Gateway 接口 Adapter
    /// </summary>
    public class SmsGatewayAdapter : SmsGateway
    {
        private static readonly ILog logger = LogManager.GetLogger(typeof(SmsGatewayAdapter));

        public async Task SendCode(string mobile, string code)
        {
            await Task.Delay(1000);
            logger.Info($"发送手机验证码 {code} 到 {mobile}");
        }
    }
}
